### PR TITLE
Translate backend validation errors

### PR DESCRIPTION
**Pull request by blacksponge**
## Purpose

Translate backend validation errors

## Context

Depends on https://github.com/zonemaster/zonemaster-backend/pull/891, addresses #239

## Changes

* Add language in param for start_domain_test and fetch_data_from_parent_zone
* Resend form if the form has been touched by the user and is in a invalid state

## How to test this PR

* Fill garbage in the test form and submit it.
* Check that errors are displayed in the selected language (if supported by the backend)
* Change the language
* Check that the errors have also been translated

### CODE REVIEW

**Review by matsduf — APPROVED** [review]
In general this looks good.

There are some cases where you change language when you get a language mix, but I think we can live with it as of now.

I approve.

**Review by matsduf — APPROVED** [review]

File: src/app/components/form/form.component.ts [hunk 1]
```diff
@@ -221,7 +233,7 @@ export class FormComponent implements OnInit, OnChanges {
     }
   }
 
-  public runDomainCheck() {
+  public runDomainCheck(submitNotValid = false) {
```
Comment (mattias-p) [inline src/app/components/form/form.component.ts:236]: Maybe give the argument a positive name?

File: src/app/components/form/form.component.ts [hunk 2]
```diff
@@ -250,7 +262,7 @@ export class FormComponent implements OnInit, OnChanges {
         digest: ds.digest
       }});
 
-    if (this.newForm.valid) {
+    if ((!submitNotValid && this.newForm.valid) || (submitNotValid && !this.newForm.valid)) {
```
Comment (mattias-p) [inline src/app/components/form/form.component.ts:265]: Could this be expressed as `submitNotValid != this.newForm.valid`?

**Review by mattias-p — DISMISSED** [review]

**Review by matsduf — DISMISSED** [review]
I approve this, and will re-approve if the comments by @mattias-p are considered.

File: src/app/components/form/form.component.ts (hunk 1 above)
Comment (blacksponge) [reply src/app/components/form/form.component.ts]: I inverted the state so it has a positive name

File: src/app/components/form/form.component.ts (hunk 2 above)
Comment (blacksponge) [reply src/app/components/form/form.component.ts]: changed

**Review by matsduf — APPROVED** [review]
I approve.

**Review by mattias-p — APPROVED** [review]

**Comment by mattias-p** [comment]
I tested this for v2021.2. It works for Swedish and French, which are the only ones currently supported in Backend. So, LGTM.

**Pull request merged**